src/components/VariantSearch/VariantSearch.tsx

```diff
@@ -124,50 +130,146 @@ export const VariantSearch = ({ onVariantSelect }: Props) => {
     return suggestions.slice(0, 20);
   };
 
+  const handleInputChange = (newValue: string, change: InputActionMeta) => {
+    if(change.action === 'input-change') {
+      // when input value has "," in the string
+      if (newValue.includes(",")) {
+        handleCommaSeparatedInput(newValue);
+      } else {
+        setInputValue(newValue);
+        setMenuIsOpen(true);
+      }
+    }
+
+    if(change.action === 'menu-close' || change.action === 'input-blur'){
+      setMenuIsOpen(false);
+    }
+  };
+
+  const handleCommaSeparatedInput = (inputValue: string) => {
+    const inputValues = inputValue.split(",")
+    let newSelectedOptions: SearchOption[] = []
+    let invalidQueries = ''
+    for (let query of inputValues) {
+      // remove spaces from both ends of the query
+      query = query.trim()
+      const suggestions = suggestOptions(query);
+      const selectedOption = suggestions.find(option => option.value.toUpperCase() === query.toUpperCase());
+      // If query gets back valid a valid suggest option which is not yet selected,
```

Comment:
Might be useful to have high level explanation comment on top of the function, rather than details inserted throughout. But this is just a styling thing, so if you prefer like this, no problem.

Reply:
Good suggestions! I added function doc and removed the inline comments.